fenix/app/src/androidTest/java/org/mozilla/fenix/ui/robots/NotificationRobot.kt

```diff
@@ -29,44 +29,48 @@ class NotificationRobot {
         var notificationFound = mDevice.findObject(notification).waitForExists(waitingTime)
 
         while (!notificationFound) {
+            Log.i(TAG, "verifySystemNotificationExists: Waiting for $waitingTime ms for notification: $notification to exist")
             scrollToEnd()
-            Log.i(TAG, "verifySystemNotificationExists: Scrolling to the end of the notification tray")
-            Log.i(TAG, "verifySystemNotificationExists: Looking for $notificationMessage notification")
             notificationFound = mDevice.findObject(notification).waitForExists(waitingTime)
+            Log.i(TAG, "verifySystemNotificationExists: Waited for $waitingTime ms for notification: $notification to exist")
         }
 
         assertUIObjectExists(itemWithText(notificationMessage))
     }
 
     fun clearNotifications() {
-        if (clearButton.exists()) {
-            Log.i(TAG, "clearNotifications: Verified that clear notifications button exists")
-            clearButton.click()
-            Log.i(TAG, "clearNotifications: Clicked clear notifications button")
+        if (clearButton().exists()) {
+            Log.i(TAG, "clearNotifications:The clear notifications button exists")
+            Log.i(TAG, "clearNotifications: Trying to click the clear notifications button")
+            clearButton().click()
+            Log.i(TAG, "clearNotifications: Clicked the clear notifications button")
         } else {
             scrollToEnd()
-            Log.i(TAG, "clearNotifications: Scrolled to end of notifications tray")
-            if (clearButton.exists()) {
-                Log.i(TAG, "clearNotifications: Verified that clear notifications button exists")
-                clearButton.click()
-                Log.i(TAG, "clearNotifications: Clicked clear notifications button")
+            if (clearButton().exists()) {
+                Log.i(TAG, "clearNotifications:The clear notifications button exists")
+                Log.i(TAG, "clearNotifications: Trying to click the clear notifications button")
+                clearButton().click()
+                Log.i(TAG, "clearNotifications: Clicked the clear notifications button")
             } else if (notificationTray().exists()) {
+                Log.i(TAG, "clearNotifications: The notifications tray exists")
```

Comment:
"The notifications tray is still displayed"

Reply:
Done !